### PR TITLE
added the Feat/workouts branch

### PR DESCRIPTION
**Pull request by MUTABAZI-D**
added the last piece of the jigsaw which is the workouts feature , so @kzngoga  kindly review it please !!!

### CODE REVIEW

**Review by kzngoga — CHANGES_REQUESTED** [review]
@MUTABAZI-D 

- I couldn't run the app due to this error. https://prnt.sc/rDEeDtLObuzw I think the issue is that you didn't add the **workoutsReducer** in the store file

I'll review more once this issue is fixed.

**Comment by MUTABAZI-D** [comment]
yes master!!! now i think the workouts branch is ready and sorry for the inconvenience @kzngoga . so please review

File: src/components/layout/WorkoutsHeader.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        justifyContent: 'center',
+        alignItems: 'center',
+        paddingRight: '150px',
+        height: '50px',
+        boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.2)',
+        zIndex: 2,
+      }}
+    >
+      <Typography variant="h4" component={'h2'} color={'primary.main'}>
+        WORKOUTS
```
Comment (kzngoga) [inline src/components/layout/WorkoutsHeader.jsx]: Use a lowercase text here. and style it to **uppercase**.

File: src/components/workouts/AddWorkoutModal.jsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+                  endAdornment: (
+                    <InputAdornment position="end">min</InputAdornment>
+                  ),
+                }}
+                name="duration"
+                type="number"
+                {...register('duration', {
+                  required: 'Duration is required',
+                  min: {
+                    value: 1,
```
Comment (kzngoga) [inline src/components/workouts/AddWorkoutModal.jsx]: Change the **min value** to 10 minutes

File: src/components/workouts/AddWorkoutModal.jsx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+                    <InputAdornment position="end">kcal</InputAdornment>
+                  ),
+                }}
+                name="caloriesBurned"
+                type="number"
+                {...register('caloriesBurned', {
+                  required: 'Calories Burned is required',
+                  min: {
+                    value: 1,
+                    message: 'Calories burned must be at least 1 Kcal',
```
Comment (kzngoga) [inline src/components/workouts/AddWorkoutModal.jsx]: Change the **min value** to 50Kcal

File: src/components/workouts/AddWorkoutModal.jsx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+                  },
+                })}
+                error={!!errors.caloriesBurned}
+                helperText={errors.caloriesBurned?.message}
+              />
+              <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (kzngoga) [inline src/components/workouts/AddWorkoutModal.jsx]: Pls remove the **date field** you need to just pass the **today's date**.

File: src/components/workouts/WorkoutsTable.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      >
+        <Typography
+          variant="h5"
+          component={'h2'}
+          textAlign={'center'}
+          gutterBottom
+          color={'primary.contrastText'}
+          sx={{ marginLeft: 2 }}
+        >
+          WORKOUTS
```
Comment (kzngoga) [inline src/components/workouts/WorkoutsTable.jsx]: Use lowercase text!

**Review by kzngoga — CHANGES_REQUESTED** [review]
@MUTABAZI-D Pls check & fix these feedback issues:

- In the table you need to fix these issues https://prnt.sc/8iAYmLCrAUBG:
	- You need to show the user information instead of the user id only. (name & phone)
	- You also need to format the date (Show the date in this format ex: 27/06/2024)
	- The search should work on both **name** and **type** fields.
	- Pls show a text like **No workouts found!** when there are no workouts found or the search results returned no data. not like this: https://prnt.sc/S5GZHw4S9TAl
	- Also the **table results** need to be sortable for now sort by date in descending order. (meaning today's date should be displayed first)
- You also need to change the hover background for this **button** https://prnt.sc/LoCqzomNErYy the current background isn't visible. you can only reduce the opacity of the button to **0.8** when it's hovered on.
- Pls remove this field (https://prnt.sc/B5whmJ-stB8O) just send today's date when adding the workout. also remove it in the edit form
- Before deleting a **workout** show a modal for confirming the action, it should have 2 buttons. i.e: **proceed** & **cancel**
	- When **cancel** is clicked then close the modal
	- When **Proceed** is clicked the delete the record

**Comment by MUTABAZI-D** [comment]
hey master!! , i just added other info from user such as phone and name , but also corrected other requested changes . so kindly re-review it please @kzngoga

**Review by kzngoga — CHANGES_REQUESTED** [review]
@MUTABAZI-D some few things to change, before we can merge:

- Remove the UserID in the table (https://prnt.sc/07m4MREynIi5)
- This one wasn't updated:
	- You also need to change the hover background for this button [prnt.sc/LoCqzomNErYy](https://prnt.sc/LoCqzomNErYy) the current background isn't visible. you can only reduce the opacity of the button to 0.8 when it's hovered on.

**Comment by MUTABAZI-D** [comment]
made requested changes master @kzngoga . please review and merge.

**Pull request merged**